api/src/main/java/ai/djl/training/GradientCollector.java

```diff
@@ -28,7 +28,8 @@
  * and added together.
  *
  * <p>Due to limitations in most engines, the gradient collectors are global. This means that only
- * one can be used at a time. If multiple are opened, an error will be thrown.
+ * one can be used at a time. If multiple are opened, an error will be thrown. However, if the
+ * collector is specified to operate on a certain block, then multiple collectors can coexist.
```

Comment:
You can't actually have multiple (even on separate blocks) based on the engine architectures. Maybe in a future engine or if they make the training mode non-global. Or, we can just treat that as an engine limitation and then throw `UnsupportedOperationExceptions` while the test suite should expect it to work rather than expect it to fail.

Reply:
Got it. I'll remove it then. Or maybe we can put it to TODO. Here is a relavant usecase: https://stackoverflow.com/questions/70998652/djl-gradientcollector-try-with-resources-initialiser-error

Reply:
Here I was guessing: inside `PtGradientCollector(Block block)`, if the following is not called unlike that in `PtGradientCollector()`, would it allow multiple GradientCollectors to coexist?
```java
        boolean wasCollecting = isCollecting.getAndSet(true);
        if (wasCollecting) {
            throw new IllegalStateException(
                    "A PtGradientCollector is already collecting. Only one can be collecting at a"
                            + " time");
        }
```